M4/practica/controllers.js

```diff
@@ -0,0 +1,218 @@
+var clientesProviderApp = angular.module('clientesProviderApp', ['ui.bootstrap']);
+
+clientesProviderApp.config(['$httpProvider', function ($httpProvider) {
+    $httpProvider.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
+}]);
+
+clientesProviderApp.controller('ListadoCtrl', ['$scope', '$http',
+    function ($scope, $http) {
+
+        /*
+        $scope.mostrar_alert = function (tab, id) {
+            $('div#alerts-container-' + tab).children('div:not(#' + id + ')').hide().parent().children('div#' + id).show();
+            // $('div#alerts-container-' + tab).children('div').hide().parent().children('div#' + id).show();
+        };
+        */
+
+        $scope.boton_buscar_disabled = false;
+        $scope.alert_shown = 'realice-busqueda';
+
+        $scope.boton_buscar = function () {
+
+            if ($scope.boton_buscar_disabled) {
+                return;
+            }
+
+            $scope.boton_buscar_disabled = true;
+            $scope.alert_shown = 'comprobando-disponibilidad';
+
+            $http.get('http://ccvalsina.com/cursoweb/autorizacion.php', {cache: false}
+            ).error(function (data) {
+                if (data.status == 401) {
+                    $scope.alert_shown = 'ocupado';
+                    $scope.boton_buscar_disabled = false;
+                } else {
+                    $scope.alert_shown = 'disponible';
+                    $scope.pedir_clientes();
+                }
+            }).success(function () {
+                $scope.alert_shown = 'disponible';
+                $scope.pedir_clientes();
+            });
+        };
+
+        $scope.clientes = [];
+
+        $scope.pedir_clientes = function () {
+
+            var impar_encontrado = 1;
+            var par_encontrado = 2;
+
+            for (var i = $scope.clientes.length - 1; i >= 0; i--) {
```

Comment:
Si, podria tener 2 tablas de clientes, separadas por paridad, y eliminar el filtro en la vista que los dividia. Igualmente, pensa que este for no se ejecuta seguido (solo cuando se buscan clientes), y tampoco por muchas iteraciones, porque al ejecutarse del final hacia adelante, los valores buscados suelen estar entre las primeras iteraciones, salvo el caso raro en el que haya muchos clientes de una misma paridad al final de la lista (muchos pares con numero alto y pocos impares con numero alto)